wordpress-mobile/WordPress-Android
# [Dependency Updates] Remove Custom Emoji Support

Parent #17565

This PR is about removing the no longer necessary custom emoji support from this project. Since this project is now on `Appcompat` version `1.4.2` (see [here](https://github.com/wordpress-mobile/WordPress-Android/issues/17561)), there is no need for custom emoji support anymore. Using `Appcompat` version `1.4.0` and above automatically adds support for latest emoji.

For more info see:
- [Use AppCompat to support the latest emoji](https://developer.android.com/develop/ui/views/text-and-emoji/emoji2#appcompat)

This change effectively reverts [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12045).

For more specifics on this change see [GitHub discussion](https://github.com/wordpress-mobile/WordPress-Android/issues/17565#issuecomment-1403805689). Cc @mzorz 

-----

PS: @thomashorta I added you as the main reviewer, that is, in addition to the @wordpress-mobile/apps-infrastructure team itself, but randomly, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid.

-----

To test:

First, follow the below set of steps building the app from `trunk`, launch it on a device running `Android 10` or lower:
1. Create a new `Story post` and insert the below set of emojis. Verify that all emojis are displayed as expected, that is, instead of a tofu (☐) emoji, or another combination of emojis. Then, publish that story and verify the emojis are also displayed as expected.
2. Using the `Block Editor`, create a new `Blog post` and insert the below set of emojis. The emojis should **NOT** be displayed as expected (Cc @fluiddot). Instead a tofu (☐) emoji, or a combination of emojis is displayed. Then, publish that post and now verify the emojis are displayed as expected, that is of course, expect when you try to edit into the post again.
3. Using the `Classic Editor`, create a new `Blog post` and insert the below set of emojis. Verify that all emojis are displayed as expected, that is, instead of a tofu (☐) emoji, or a combination of emojis. Then, publish that post and verify the emojis are also displayed as expected, even when you try to edit the post again.
4. Navigate to the below screens that are somewhat utilizing some kind of emoji related functionality and verify that all emojis are displaying as expected:
    - Site Creation Screen (Go to `Site Picker`, click the `+` button at the top right and finally select `Create WordPress.com site`)
    - Layout Picker Screen (Create a new `Page` via the FAB, this will navigate you to that screen where you could choose a layout)
    - Comments Screen (On the `My Side` main tab, find the `MENU` inner tab and its `Publish` section, click on the `Comments` item)
5. Smoke test both, the WordPress and Jetpack apps, any other way and verify that all emojis are displayed as expected.

When you done with the above, repeat the exact same steps, this building the app from this `branch`:

FYI: Set of emojis per version (as copy-pasted from the [docs](https://developer.android.com/develop/ui/views/text-and-emoji/emoji2#appcompat)):
- ❌ 14.0: 🫠, 🫱🏼‍🫲🏿, 🫰🏽
- ✅ 13.1: 😶‍🌫️, 🧔🏻‍♀️, 🧑🏿‍❤️‍🧑🏾
- ✅ 13.0: 🥲, 🥷🏿, 🐻‍❄️
- ✅ 12.1: 🧑🏻‍🦰, 🧑🏿‍🦯, 👩🏻‍🤝‍👩🏼
- ✅ 12.0: 🦩, 🦻🏿, 👩🏼‍🤝‍👩🏻

PS: The only sets of emojis that shouldn't be displayed as expected are the newer [14.0](https://emojipedia.org/emoji-14.0/) version related emojis ones.

-----

## Regression Notes

1. Potential unintended areas of impact

    - New emojis that should have been supported on Android 10 or lower devices via the `Appcompat` library are not displaying as expected. For example, the [14.0](https://emojipedia.org/emoji-14.0/) version related emojis (ie. the 🫠 , 🫱🏼‍🫲🏿 , 🫰🏽) are not displaying correctly anyway, and this is known as it was confirmed during testing, but emojis as of [13.1](https://emojipedia.org/emoji-13.1/) and before should be displayed correctly.

7. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

8. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
